h2/src/main/org/h2/mode/FunctionsPostgreSQL.java

```diff
@@ -221,6 +231,27 @@ protected Value getValueWithArgs(Session session, Expression[] args) {
             // Not implemented
             result = v1.convertTo(Value.VARCHAR);
             break;
+        case ARRAY_TO_STRING:
+            StringJoiner joiner = new StringJoiner(v1.getString());
+            if (v0.getValueType() != Value.ARRAY) {
+                throw DbException.getInvalidValueException("ARRAY_TO_STRING array", v0);
+            }
+            if (v2 != null) {
+                int v2Type = v2.getValueType();
+                if (v2Type < Value.CHAR || v2Type > Value.VARCHAR_IGNORECASE) {
+                    throw DbException.getInvalidValueException("ARRAY_TO_STRING optional_null_string", v2);
+                }
```

Comment:
Third argument may also be `NULL`. I think there is no reason to bother with its data type, you can use something like that:
```Java
String nullValue = null;
if (v2 != null) {
    nullValue = v2.getString();
}
```

Reply:
Fixed.

In real PostgreSQL 12, however, 3rd arg can only be NULL or String-like.